### ISSUE TITLE
Laravel 5.3 Support

### ISSUE THREAD

**Issue by win45** [auth0/laravel-auth0#45]

**Comment by golinmarq** [comment]
What should I do to use this in Laravel 5.3?

**Comment by glena** [comment]
Didn't have time to work on it yet. I will focus on this on the upcoming weeks.

**Comment by glena** [comment]
I just tested it and should work fine on laravel 5.3.

The only difference is how to enable this on `config/auth0.php`. You only need to change the providers configuration like this.

```
    'providers' => [
        'users' => [
            'driver' => 'auth0'
        ],
    ],
```

The rest was the same than here https://auth0.com/docs/quickstart/webapp/laravel#6-1-using-the-auth0-driver

**Comment by carnevalle** [comment]
I am also trying to use Auth0 with Laravel 5.3. 

I am using the eloquent driver and persisting the user to the database. I have problem with logging out. I am mentioning it here, as I don't know if this is related to Laravel 5.3.

The docs says that you should do this to log out:
`
Auth::logout();
`

When doing this, I would get a previous user when trying to log in with another user. So I had to do this to clear the previous user when logging out:
`Auth::logout();
Auth0::logout();`

Am I doing something wrong - or should this be added to the docs?

**Comment by glena** [comment]
I am about to release the v4.0.0 which support auth0-php 4.0.0 (there are some changed in the API of the vanilla php SDK) which should fix it.

Can you try the following to confirm this is working correctly? It would be really usefull to confirm before the release.

Can you hook to this three events:

```php
        \Event::listen('auth.logout', function() {
            \App::make("auth0")->logout();
        });
        \Event::listen('user.logout', function() {
            \App::make("auth0")->logout();
        });
        \Event::listen('Illuminate\Auth\Events\Logout', function() {
            \App::make("auth0")->logout();
        });
```

Laravel keeps changing things and usually there is no mention of the changes in the docs. I think the event changed from `auth.logout` to `user.logout` to `Illuminate\Auth\Events\Logout` (I will keep the three of them for BC).

**Comment by carnevalle** [comment]
The above seems to work 👍

Great to switch event names without notice :-)

**Comment by glena** [comment]
I released `3.2.1` today with that fix. if you update the dependency it should work :D
